### ISSUE TITLE
Problem with cache and large-ish figures

### ISSUE THREAD

**Issue by schrimpf** [mpastell/Weave.jl#157]
Attempting to use cache with figures whose stringmime() is more than 2^16 characters throws an error. For example, with the following test.jmd file
```julia
```julia, cache=true
using Plots, LinearAlgebra
Plots.gr()
n = 100
x = rand(n)
y = rand(n)
xg=0:0.02:1
scatter(x,y)
f = (x,y) -> norm([x,y])
contour!(xg,xg,f)
````
The following error occurs
```julia
julia> using Weave; weave("test.jmd", cache=:refresh)
[ Info: Weaving chunk 1 from line 2
Error encountered while saving "/home/paul/ECON628_2018/paul/02-extremumEstimation/cache/test.jld2".
Fatal error:
ERROR: InexactError: trunc(UInt16, 88250)
Stacktrace:
 [1] handle_error(::InexactError, ::FileIO.File{FileIO.DataFormat{:JLD2}}) at /home/paul/.julia/packages/FileIO/Y0Lre/src/error_handling.jl:80
 [2] handle_exceptions(::Array{Any,1}, ::String) at /home/paul/.julia/packages/FileIO/Y0Lre/src/error_handling.jl:75
 [3] #save#30(::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}, ::Function, ::FileIO.File{FileIO.DataFormat{:JLD2}}, ::Dict{String,Weave.WeaveDoc}) at /home/paul/.julia/packages/FileIO/Y0Lre/src/loadsave.jl:217
 [4] save(::FileIO.File{FileIO.DataFormat{:JLD2}}, ::Dict{String,Weave.WeaveDoc}) at /home/paul/.julia/packages/FileIO/Y0Lre/src/loadsave.jl:199
 [5] #save#15(::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}, ::Function, ::String, ::Dict{String,Weave.WeaveDoc}) at /home/paul/.julia/packages/FileIO/Y0Lre/src/loadsave.jl:118
 [6] save(::String, ::Dict{String,Weave.WeaveDoc}) at /home/paul/.julia/packages/FileIO/Y0Lre/src/loadsave.jl:118
 [7] #invokelatest#1 at ./essentials.jl:697 [inlined]
 [8] invokelatest at ./essentials.jl:696 [inlined]
 [9] write_cache(::Weave.WeaveDoc, ::String) at /home/paul/.julia/packages/Weave/1O5IK/src/cache.jl:6
 [10] #run#21(::Symbol, ::Symbol, ::Symbol, ::Dict{Any,Any}, ::String, ::Nothing, ::String, ::Symbol, ::Bool, ::typeof(run), ::Weave.WeaveDoc) at /home/paul/.julia/packages/Weave/1O5IK/src/run.jl:79
 [11] #run at ./none:0 [inlined]
 [12] #weave#10(::Symbol, ::Symbol, ::Symbol, ::Dict{Any,Any}, ::Symbol, ::String, ::Nothing, ::String, ::Symbol, ::Bool, ::Nothing, ::Nothing, ::Nothing, ::Array{String,1}, ::String, ::typeof(weave), ::String) at /home/paul/.julia/packages/Weave/1O5IK/src/Weave.jl:107
 [13] (::getfield(Weave, Symbol("#kw##weave")))(::NamedTuple{(:cache,),Tuple{Symbol}}, ::typeof(weave), ::String) at ./none:0
 [14] top-level scope at none:0
 [15] run_backend(::REPL.REPLBackend) at /home/paul/.julia/packages/Revise/TmjcT/src/Revise.jl:766
 [16] (::getfield(Revise, Symbol("##58#60")){REPL.REPLBackend})() at ./task.jl:259
```

You can produce essentially the same error by just trying to save a long string with JLD2
```
longstring = ""
for i in 1:(2^16)
  global longstring *= "a"
end
using FileIO, JLD2
FileIO.save("test.jld2", Dict("fig" => longstring))
```
I don't know whether this is intended behavior for JLD2 or a bug there.

**Comment by tlnagy** [comment]
This is fixed by #158, right?

**Comment by mpastell** [comment]
Yes!
